hotsos/core/plugins/kernel/net.py

```diff
@@ -220,3 +220,81 @@ def _fields(self):
             'TCPReqQFullDoCookies',
             # skb retrans before original left host
             'TCPSpuriousRtxHostQueues']
+
+
+class ProcSockStat():
+
+    """
+    Provides a common way to extract fields from /proc/net/sockstat.
+
+    Expected file format is:
+
+    hdr1: label1 value1 label2 value2...
+    hdr2: label1 value1 label2 value2...
+    hdr3: label1 value1 label2 value2...
+    """
+    def __init__(self):
+        self._data = {}
+        self._process_file(os.path.join(HotSOSConfig.DATA_ROOT,
+                                        'proc/net/sockstat'))
+
+    def _process_file(self, fname):
+        if not os.path.exists(fname):
+            log.info("file not found '%s' - skipping load")
+            return
+
+        log.debug("start processing %s", fname)
+        with open(fname) as fd:
+            for line in fd:
+                psl = line.split(':')
+                if len(psl) != 2:
+                    log.warning("badness: line does not match format %s", line)
+                    continue
+                (label, stats) = psl
+                if label not in self._data:
+                    self._data[label] = {}
+                try:
```

Comment:
Need to make sure integers are saved as int() so that they can be compared with integers.